### PR TITLE
bnxt_hwrm_set_filter() take a dst_id instead of a vnic

### PR DESCRIPTION
**Pull request by stephen-hurd**
Support source type/id in filters
Only call FUNC_VF_VNIC_IDS_QUERY from bnxt_hwrm_func_vf_vnic_query()
Make bnxt_hwrm_func_vf_vnic_query() static

### CODE REVIEW

File: drivers/net/bnxt/bnxt_hwrm.c [hunk 1]
```diff
@@ -2473,6 +2497,8 @@ int bnxt_hwrm_func_vf_vnic_query_and_config(struct bnxt *bp, uint16_t vf,
 		rc = bnxt_hwrm_vnic_qcfg(bp, &vnic, bp->pf.first_vf_id + vf);
 		if (rc)
 			break;
+		if (vnic.mru == 4)	// Unallocated
+			continue;
```
Comment (ajitkhaparde) [inline drivers/net/bnxt/bnxt_hwrm.c:2501]: Is this related to this fix?

File: drivers/net/bnxt/bnxt_hwrm.c [hunk 2]
```diff
@@ -285,7 +285,7 @@ int bnxt_hwrm_set_filter(struct bnxt *bp,
 
 	enables = filter->enables |
 	      HWRM_CFA_L2_FILTER_ALLOC_INPUT_ENABLES_DST_ID;
-	req.dst_id = rte_cpu_to_le_16(vnic->fw_vnic_id);
+	req.dst_id = rte_cpu_to_le_16(dst_id);
```
Comment (ajitkhaparde) [inline drivers/net/bnxt/bnxt_hwrm.c:288]: This is essentially the same. Right?
Previously we were passing vnic, but just dereferencing fw_vnic_id from that.
Now we pass just the vnic->fw_vnic_id.

File: drivers/net/bnxt/bnxt_hwrm.c [hunk 3]
```diff
@@ -301,6 +301,10 @@ int bnxt_hwrm_set_filter(struct bnxt *bp,
 	if (enables &
 	    HWRM_CFA_L2_FILTER_ALLOC_INPUT_ENABLES_L2_OVLAN_MASK)
 		req.l2_ovlan_mask = filter->l2_ovlan_mask;
+	if (enables & HWRM_CFA_L2_FILTER_ALLOC_INPUT_ENABLES_SRC_ID)
+		req.src_id = rte_cpu_to_le_32(filter->src_id);
+	if (enables & HWRM_CFA_L2_FILTER_ALLOC_INPUT_ENABLES_SRC_TYPE)
+		req.src_type = rte_cpu_to_le_32(filter->src_type);
```
Comment (ajitkhaparde) [inline drivers/net/bnxt/bnxt_hwrm.c]: Are these helping? Just curious..
But we are not setting any of the ENABLES_SRC_ID/TYPE yet?

File: drivers/net/bnxt/bnxt_hwrm.c (hunk 2 above)
Comment (stephen-hurd) [reply drivers/net/bnxt/bnxt_hwrm.c:288]: Right.  Not all dst_id values are VNIC IDs though.  This allows set_filter to set TX filters and such as well now.

File: drivers/net/bnxt/bnxt_hwrm.c (hunk 3 above)
Comment (stephen-hurd) [reply drivers/net/bnxt/bnxt_hwrm.c]: It appears we don't need them for now, but we likely will soon.

File: drivers/net/bnxt/bnxt_hwrm.c (hunk 1 above)
Comment (stephen-hurd) [reply drivers/net/bnxt/bnxt_hwrm.c:2501]: There were some firmware issues with us doing anything with unallocated VNICs.  This prevents that from happening.

File: drivers/net/bnxt/bnxt_hwrm.c (hunk 3 above)
Comment (ajitkhaparde) [reply drivers/net/bnxt/bnxt_hwrm.c]: The HWRM src_type is a u8. We don't need the cpu_to_le32.

Comment (ajitkhaparde) [reply drivers/net/bnxt/bnxt_hwrm.c]: Old code.. the l2_ovlan_mask and l2_ovlan are u16. but we are not doing the cpu_tole16 on the assigned values.
Do you want to fix that as well?

Comment (stephen-hurd) [reply drivers/net/bnxt/bnxt_hwrm.c]: Yeah, I'll get all of those issues, thanks.

**Comment by stephen-hurd** [comment]
Actually, those vlan items should be in network byte order, so shouldn't be touched.

File: drivers/net/bnxt/bnxt_hwrm.c (hunk 1 above)
Comment (ajitkhaparde) [reply drivers/net/bnxt/bnxt_hwrm.c:2501]: unrelated to the dst_id change then. But nee how. Its ok.

**Review by ajitkhaparde — APPROVED** [review]
